src/Mentat.Domain/Service/StudentService.cs

```diff
@@ -19,12 +19,14 @@ ICardService cardService
         public StudentVM GetStudentVM(List<string> selectedDifficulties)
         {
             var cards = _cardService.GetFilteredCardsList(selectedDifficulties);
+            var tags = _cardService.GetAllTags();
             var vm = new StudentVM
             {
                 AvailableCards = new List<FlashCardVM>(),
                 SelectedCardIndex = 1,
                 FilterCount = selectedDifficulties?.Count ?? 0,
-                SelectedDifficulties = selectedDifficulties
+                SelectedDifficulties = selectedDifficulties,
+                AllUniqueTags = tags
```

Comment:
Are tags guaranteed to be unique when they are read from the database?

Reply:
I think so. tags is a List<string> that is populated by GetAllTags(). It gets all flashcard documents, loops through them one by one, and adds any tags to a HashSet in order to ensure uniqueness. Then it converts the HashSet to a List.
```csharp 
            HashSet<string> uniqueTags = new HashSet<string>();
            foreach (var doc in documents)
            {
                if (doc.Tags != null)
                    uniqueTags.UnionWith(doc.Tags);
            }
            return uniqueTags.ToList();
```